substrate/frame/staking/stake-tracker/src/lib.rs

```diff
@@ -0,0 +1,420 @@
+		let nominator_vote = Self::weight_of(Self::active_vote_of(who));
+
+		log!(
+			debug,
+			"on_nominator_remove: {:?} with {:?}. impacting {:?}",
+			who,
+			nominator_vote,
+			nominations,
+		);
+
+		// updates the nominated target's score. Note: this may update the score of up to
+		// `T::MaxNominations` validators.
+		for t in nominations.iter() {
+			Self::update_score::<T::TargetList>(&t, StakeImbalance::Negative(nominator_vote.into()))
+		}
+
+		let _ = T::VoterList::on_remove(&who).defensive_proof(
+			"the nominator exists in the list as per the contract with staking; qed.",
+		);
+	}
+
+	// Fired when someone removes their intention to validate, either due to chill or nominating.
+	fn on_validator_remove(who: &T::AccountId) {
+		log!(debug, "on_validator_remove: {:?}", who,);
+
+		let _ = T::TargetList::on_remove(&who).defensive_proof(
+			"the validator exists in the list as per the contract with staking; qed.",
+		);
+
+		// validator is also a nominator.
+		Self::on_nominator_remove(who, vec![]);
+	}
+
+	// Fired when an existing nominator updates their nominations.
+	//
+	// This is called when a nominator updates their nominations. The nominator's stake remains the
+	// same (updates to the nominator's stake should emit [`Self::on_stake_update`] instead).
+	// However, the score of the nominated targets must be updated accordingly.
+	//
+	// Note: it is assumed that who's staking state is updated *before* calling this method.
+	fn on_nominator_update(who: &T::AccountId, prev_nominations: Vec<T::AccountId>) {
+		let nominator_vote = Self::weight_of(Self::active_vote_of(who));
+
+		let curr_nominations =
+			<T::Staking as StakingInterface>::nominations(&who).unwrap_or_default();
+
+		log!(
+			debug,
+			"on_nominator_update: {:?}, with {:?}. previous nominations: {:?} -> new nominations {:?}",
+			who, nominator_vote, prev_nominations, curr_nominations,
+		);
+
+		// new nominations
+		for target in curr_nominations.iter() {
+			if !prev_nominations.contains(target) {
+				Self::update_score::<T::TargetList>(
+					&target,
+					StakeImbalance::Positive(nominator_vote.into()),
+				);
+			}
+		}
+		// removed nominations
+		for target in prev_nominations.iter() {
+			if !curr_nominations.contains(target) {
+				Self::update_score::<T::TargetList>(
+					&target,
+					StakeImbalance::Negative(nominator_vote.into()),
+				);
+			}
+		}
+	}
+
+	// noop: the updates to target and voter lists when applying a slash are performed
+	// through [`Self::on_nominator_remove`] and [`Self::on_validator_remove`] when the stakers are
+	// chilled. When the slash is applied, the ledger is updated of the affected stashes is, thus
+	// the stake is propagated through the `[Self::update::<T::EventListener>]`.
+	fn on_slash(
+		_stash: &T::AccountId,
+		_slashed_active: BalanceOf<T>,
+		_slashed_unlocking: &BTreeMap<sp_staking::EraIndex, BalanceOf<T>>,
+		_slashed_total: BalanceOf<T>,
+	) {
+	}
```

Comment:
```suggestion
	/// noop: the updates to target and voter lists when applying a slash are performed
	/// through [`Self::on_nominator_remove`] and [`Self::on_validator_remove`] when the stakers are
	/// chilled. When the slash is applied, the ledger is updated of the affected stashes is, thus
	/// the stake is propagated through the [`Self::update::<T::EventListener>`].
	fn on_slash(
		_stash: &T::AccountId,
		_slashed_active: BalanceOf<T>,
		_slashed_unlocking: &BTreeMap<sp_staking::EraIndex, BalanceOf<T>>,
		_slashed_total: BalanceOf<T>,
	) {
	}
```

Reply:
I think that the comments here should not be rustdoc, it's just a note to the developer. wdyt? in any case, added the correction to the [\`Self::update::<T::EventListener>\`]. manually, thanks!

Reply:
why not make it a note to the developer in the form of a rustdoc? otherwise the method is there with no documentation